tests/candev/Makefile

```diff
@@ -4,14 +4,21 @@ USEMODULE += shell
 USEMODULE += can
 USEMODULE += isrpipe
 
-FEATURES_OPTIONAL += periph_can
+#native board only supports periph_can
+ifeq ($(BOARD), native)
+	CAN_DRIVER = PERIPH_CAN
+endif
 
-# define the CAN driver you want to use here
-CAN_DRIVER ?= CAN_EXT
 
-ifeq ($(CAN_DRIVER), CAN_EXT)
-# external CAN driver modules/variables go here
+# define the CAN driver you want to use here
+CAN_DRIVER ?= MCP2515
 
+ifeq ($(CAN_DRIVER), PERIPH_CAN)
+	FEATURES_REQUIRED += periph_can
+else ifeq ($(CAN_DRIVER), MCP2515)
+	USEMODULE += mcp2515
+else ifeq ($(CAN_DRIVER), CAN_ALT)
+	# other can modules can be defined here
 endif
```

Comment:
I think I never explained why we had this `Makefile.board.dep`. The idea is that in the application `Makefile` you would only have dependencies that do not depend on the target `CPU`/`BOARD`, etc. This is because the ci decides what to build for which BOARD based on `info-boards-supported` for that target application. This recipe is executed after parsing the application `Makeifle`. The issue with having platform dependencies in the application Makefile is that the result will depend on what the default BOARD is, the workaroud for this is havig platform specific dependencies for an application in `Makefile.board.dep`. There are more details on this in #12755 (PR that introduced this).

Concretely this means that because of 

```
#native board only supports periph_can
ifeq ($(BOARD), native)
  CAN_DRIVER = PERIPH_CAN
endif
```

Then all this:

```
#native board only supports periph_can
ifeq ($(BOARD), native)
  CAN_DRIVER = PERIPH_CAN
endif

# define the CAN driver you want to use here
CAN_DRIVER ?= MCP2515

ifeq ($(CAN_DRIVER), PERIPH_CAN)
  FEATURES_REQUIRED += periph_can
else ifeq ($(CAN_DRIVER), MCP2515)
  USEMODULE += mcp2515
else ifeq ($(CAN_DRIVER), CAN_ALT)
  # other can modules can be defined here
endif
```

needs to go in `Makefile.board.dep`

I know this is not straight forward, but without this since the default platform is native then `mcp2515` would never be built by the CI. But yes I agree that it hides things a bit, but this is the way to format test applications for now :( .

Reply:
Ok, apologies for that. I'm not fully familiar with all the constructs used in RIOT. Especially the ones related to the build-process. Thanks for the explanation.

I removed the check in the Makefile and updated the README accordingly. I did add a note that periph_can is to be used when building for native though.

Reply:
> Ok, apologies for that. I'm not fully familiar with all the constructs used in RIOT. Especially the ones related to the build-process. 

No worries, the build system is definitively the most confusing part in RIOT, thanks for the patch!